server/src/main/java/p2/webservice/FavoriteWebService.java

```diff
@@ -0,0 +1,123 @@
+package p2.webservice;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+
+import p2.dao.impl.FavoriteDAO;
+import p2.model.Favorite;
+import p2.model.Product;
+import p2.model.User;
+import p2.service.impl.FavoriteService;
+import p2.service.impl.ProductService;
+import p2.service.impl.UserService;
+
+public class FavoriteWebService {
+	
+
+	public static void insert(HttpServletRequest request, HttpServletResponse response) {
+		int UID = Integer.parseInt(request.getParameter("customerid"));
+		int PID = Integer.parseInt(request.getParameter("productid"));
+
+		User u = UserService.findById(UID);
+		Product p = ProductService.findById(PID);
+		
+		Favorite f = new Favorite();
+		f.setUser(u);
+		f.setProduct(p);
+		
+		
+
+		FavoriteService.insert(f);
+		//will need to change some voids to booleans for a return value
+//		ObjectMapper om = new ObjectMapper();
+//		try {
+//			String json = om.writeValueAsString(f);
+//			response.getWriter().append(json).close();
+//		} catch (IOException e) {
+//			e.printStackTrace();
+//		}
+		
+	}
+
+
+	public static void update(HttpServletRequest request, HttpServletResponse response) {
+		int FID = Integer.parseInt(request.getParameter("favoriteid"));
```

Comment:
these will throw InputMismatchException on request.getParameter returning null (which might happen if the api is called while testing)
also, we can use the new Product, User and Favorite model constructors to save a few api calls here

Reply:
I dont think we actually need the user and favorite records here, just the ids. Hibernate will just be grabbing the ids from user and favorite anyway to insert or update a row in the table